syntax = "proto2";

import "lib/common.proto";

message SectorMappingProto {
	message MappingProto {
		optional int32 logical = 1 [(help) = "logical sector ID"];
		optional int32 physical = 2 [(help) = "physical sector ID"];
	}

	message TrackdataProto {
		optional int32 track = 1       [(help) = "if present, this mapping only applies to this track"];
		optional int32 up_to_track = 2 [(help) = "if present, forms a range with track"];
		optional int32 side = 3        [(help) = "if present, this mapping only applies to this side"];

		repeated MappingProto mapping = 4 [(help) = "the sector mapping"];
	}

	repeated TrackdataProto trackdata = 1 [(help) = "per-track sector mapping"];
}

